PurchasesCoreSwift/Networking/Backend.swift

```diff
@@ -0,0 +1,664 @@
+        let unknownEligibilityClosure: () -> [String: IntroEligibility] = {
+            let unknownEligibilities = [IntroEligibility](repeating: IntroEligibility(eligibilityStatus: .unknown),
+                                                          count: productIdentifiers.count)
+            let productIdentifiersToEligibility = zip(productIdentifiers, unknownEligibilities)
+            return Dictionary(uniqueKeysWithValues: productIdentifiersToEligibility)
+        }
+
+        guard let appUserID = try? escapedAppUserID(appUserID: appUserID) else {
+            completion(unknownEligibilityClosure(), ErrorUtils.missingAppUserIDError())
+            return
+        }
+
+        let fetchToken = receiptData.rc_asFetchToken
+        let path = "/subscribers/\(appUserID)/intro_eligibility"
+        let body: [String: Any] = ["product_identifiers": productIdentifiers,
+                                   "fetch_token": fetchToken]
+
+        httpClient.performPOSTRequest(serially: true,
+                                      path: path,
+                                      requestBody: body,
+                                      headers: authHeaders) { statusCode, maybeResponse, error in
+            let eligibilityResponse = IntroEligibilityResponse(maybeResponse: maybeResponse,
+                                                               statusCode: statusCode,
+                                                               error: error,
+                                                               productIdentifiers: productIdentifiers,
+                                                               unknownEligibilityClosure: unknownEligibilityClosure,
+                                                               completion: completion)
+            self.handleIntroEligibility(response: eligibilityResponse)
+        }
+    }
+
+}
+
+private extension Backend {
+
+    enum CallbackCacheStatus {
+
+        // When an array exists in the cache for a particular path, we add to it and return this value.
+        case addedToExistingInFlightList
+
+        // When an array doesn't yet exist in the cache for a particular path, we create one, add to it
+        // and return this value.
+        case firstCallbackAddedToList
+
+    }
+
+    func handleIntroEligibility(response: IntroEligibilityResponse) {
+        var responseData = response.maybeResponse
+        if response.statusCode >= HTTPStatusCodes.redirect.rawValue || response.error != nil {
+            responseData = [:]
+        }
+
+        guard let responseData = responseData else {
+            response.completion(response.unknownEligibilityClosure(), nil)
+            return
+        }
+
+        var eligibilities: [String: IntroEligibility] = [:]
+        for productID in response.productIdentifiers {
+            let status: IntroEligibilityStatus
+
+            if let e = responseData[productID] as? Bool {
+                status = e ? .eligible : .ineligible
```

Comment:
I'd still name this `eligibility` because I'm not a huge fan of single-letter variables

Reply:
✅